### ISSUE TITLE
PlatformException after updating the application in google play store.

### ISSUE THREAD

**Issue by NikitaMasev** [mogol/flutter_secure_storage#193]
I publish a second update to the app using this package. User data is stored there. And after the update occurs, when trying to read, we get an error:

PlatformException(Exception encountered, read, java.lang.NullPointerException: Attempt to invoke interface method 'bytel]
com.it _nomads.fluttersecurestorage.ciphers.StorageC ipher.decrypt(byte[)' on a null object reference at com.it_nomads.fluttersecurestorage.FlutterSecureStor agePlugin.decodeRawValue(FlutterSecureStoragePlug in.java:162) at com.it_nomads.fluttersecurestorage.FlutterSecureStor agePlugin.read(FlutterSecureStoragePlugin.java:143) at com.it_nomads.fluttersecurestorage.FlutterSecureStor agePlugin.access$300(FlutterSecureStoragePlugin.jav a:30) at com.it_nomads.fluttersecurestorage.FlutterSecureStor agePlugin$MethodRunner.run(FlutterSecureStoragePl ugin.java:196) at android.os.Handler.handleCallback(Handler.java:938) at android.os.Handler.dispatchMessage(Handler.java:99) at android.os.Looper.loop(Looper.java:223) android.os.HandlerThread.run(HandlerThread.java:67) , null)

This error cannot be caught in the debug version of the application or even when running the command flutter run --release. This error can be caught only after it has been published in Google Play.

flutter doctor -v
![image](https://user-images.githubusercontent.com/22840281/102880536-8a208880-445c-11eb-911f-658d28bd9b38.png)

**Comment by jjsebastianfuertes** [comment]
How are you using it?  can you please provide an example of your code where this platform exception jumps up;

**Comment by NikitaMasev** [comment]
@jjsebastianfuertes Here some example code.
`
const AUTH = 'AUTH';

class UserDataSecureStorage with UserStorage {
  final FlutterSecureStorage _storage;

  UserDataSecureStorage(this._storage);

  @override
  Future<void> saveAuthUserInfo(String data) {
    return _storage.write(key: AUTH, value: data);
  }

  @override
  Future<String> getAuthUserInfo() {
    return _storage.read(key: AUTH);
  }
}
`

String data is the just json:
`
import 'package:json_annotation/json_annotation.dart';

part 'auth_user_info.g.dart';

@JsonSerializable()
class AuthUserInfo {
  String login;
  String password;
  String baseUrl;

  AuthUserInfo(this.login, this.password, this.baseUrl);

  factory AuthUserInfo.fromJson(Map<String, dynamic> json) => _$AuthUserInfoFromJson(json);

  Map<String, dynamic> toJson() => _$AuthUserInfoToJson(this);

}
`
I contacted one of the users. He was already authorized in the application, i.e. the AuthUserInfo object was saved. Then he updated the application in the google play store, deleted the application and downloaded it again. When trying to login, he was getting this error. The getAuthUserInfo () function call has been wrapped in try catch.

**Comment by abdulpiksi** [comment]
you can use [this](https://github.com/mogol/flutter_secure_storage/issues/53#issuecomment-506715829)

**Comment by Sujal1** [comment]
Has this issue been solved?

**Comment by juliansteenbakker** [comment]
I am closing this as duplicate of #53

**Comment by harSofja** [comment]
I have been receiving this error as well:

PlatformException (Exception encountered, write, java.lang.NullPointerException: Attempt to invoke interface method 'byte[] a5.i.a (bytel]) on a null object reference at z4.a.p (Unknown Source:24) at z4.e$b.run (Unknown Source:278) at android.os.Handler.handleCallback(Handler.java:969) at android.os.Handler.dispatchMessage(Handler.java:100) at android.os.Looper.loopOnce(Looper.java:201) at android.os.Looper.loop(Looper.java:288) at android.os.HandlerThread.run (HandlerThread.java:67) , null) 1


I have not being able to figure out what it is. Our app is already live and this only occurs for a specific android tablet called 'Blackview Tab 30 WiFi'. Last time I faced something similar called 'BadPaddingException' it was again on a specific android tablet. In general I have not being able to reproduce this error on any other physical device or emulators. We had managed to fix the BadPaddingException by telling the customer to do a factory reset and it worked. This is not ideal I know. Note that we did the same to this guy with the Blackview tablet but this did not fix it this time. The error happened during login phase in both cases.

I haven't been able to find much on the issue, right now I am just trying to add null checks everywhere during login phase. I am a junior dev.
